From 4eb5d3cb4aec8f1f2da9eaba11b936687024bd58 Mon Sep 17 00:00:00 2001
From: raghavt20 <raghavt20@aospa.co>
Date: Wed, 28 Oct 2026 09:32:12 +0530
Subject: [PATCH 08/11] Disable sweet hardware specific features

Violet only has a 48MP QCFA main camera next to the depth sensor, so
turn off what needs sweet's other cameras: ultra wide (config, LDC,
super night UW, lens switching, dual lens manual items), macro and
super macro, optical/SAT zoom and fallback, fast zoom-in and dual
camera recording, and report the 48MP (1:8000x6000) ultra pixel mode
instead of 108/64MP.
---
 smali_classes2/com/mi/device/Sweet.smali | 36 ++++++++++++------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/smali_classes2/com/mi/device/Sweet.smali b/smali_classes2/com/mi/device/Sweet.smali
index fd7ec4f..edc538e 100644
--- a/smali_classes2/com/mi/device/Sweet.smali
+++ b/smali_classes2/com/mi/device/Sweet.smali
@@ -159,7 +159,7 @@
 .method public o0000O0()Ljava/lang/String;
     .locals 0
 
-    const-string p0, "2:9280x6944"
+    const-string p0, "1:8000x6000"
 
     return-object p0
 .end method
@@ -244,7 +244,7 @@
 .method public o000OoO()Ljava/lang/String;
     .locals 0
 
-    const-string p0, "MACRO:WIDE:ULTRA_WIDE:SAT"
+    const-string p0, ""
 
     return-object p0
 .end method
@@ -388,7 +388,7 @@
 .method public o00O0Oo0()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -396,7 +396,7 @@
 .method public o00O0Ooo()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -524,7 +524,7 @@
 .method public o00Oo0oO()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -572,7 +572,7 @@
 .method public o00Ooo0()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -716,7 +716,7 @@
 .method public o00oO00O()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -740,7 +740,7 @@
 .method public o00oOo()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -748,7 +748,7 @@
 .method public o00oOo00()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -764,7 +764,7 @@
 .method public o00oOoO0()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -820,7 +820,7 @@
 .method public o00oo0OO()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -828,7 +828,7 @@
 .method public o00ooOoo()I
     .locals 0
 
-    const/4 p0, 0x2
+    const/4 p0, 0x1
 
     return p0
 .end method
@@ -884,7 +884,7 @@
 .method public o0O00O()Ljava/lang/String;
     .locals 0
 
-    const-string p0, "ultra_wide:macro:pro:capture_intent"
+    const-string p0, ""
 
     return-object p0
 .end method
@@ -1068,7 +1068,7 @@
 .method public o0O0oOOO()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -1092,7 +1092,7 @@
 .method public o0O0oo00()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -1156,7 +1156,7 @@
 .method public o0OoOoOo()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -1164,7 +1164,7 @@
 .method public o0o0Oo()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
@@ -1212,7 +1212,7 @@
 .method public oo0oOO0()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
-- 
2.50.1 (Apple Git-155)

